plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1263,149 @@ def max_deflection(self):
-        h = ((h + np.roll(h, -1)) / 2)[0:-1]
-        v = ((v + np.roll(v, -1)) / 2)[0:-1]
+    if bins is None:
+        bins = [200, 200]
 
-        # Throw a warning if < 50% of the particles are included on the
-        # histogram
-        percentage = np.sum(intensity) / self.nparticles
-        if percentage < 0.5:
-            warnings.warn(
-                f"Only {percentage:.2%} of the particles are shown "
-                "on this synthetic radiograph. Consider increasing "
-                "the size to include more.",
-                RuntimeWarning,
-            )
+    # Note that, at the end of the simulation, all particles were moved
+    # into the image plane.
+
+    # If ignore_grid is True, use the predicted positions in the absence of
+    # simulated fields
+    if ignore_grid:
+        xloc = d["x0"]
+        yloc = d["y0"]
+    else:
+        xloc = d["x"]
+        yloc = d["y"]
+
+    if size is None:
+        # If a detector size is not given, choose a size based on the
+        # particle positions
+        w = np.max([np.max(np.abs(xloc)), np.max(np.abs(yloc))])
+
+        # The factor of 5 here is somewhat arbitrary: we just want a
+        # region a few times bigger than the image of the grid on the
+        # detector, since particles could be deflected out
+        size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+
+    # Generate the histogram
+    intensity, h, v = np.histogram2d(xloc, yloc, range=size.to(u.m).value, bins=bins)
+
+    # h, v are the bin edges: compute the centers to produce arrays
+    # of the right length (then trim off the extra point)
+    h = ((h + np.roll(h, -1)) / 2)[0:-1]
+    v = ((v + np.roll(v, -1)) / 2)[0:-1]
+
+    # Throw a warning if < 50% of the particles are included on the
+    # histogram
+    percentage = np.sum(intensity) / d["nparticles"]
+    if percentage < 0.5:
+        warnings.warn(
+            f"Only {percentage:.2%} of the particles are shown "
+            "on this synthetic radiograph. Consider increasing "
+            "the size to include more.",
+            RuntimeWarning,
+        )
 
-        if optical_density:
-            # Generate the null radiograph
-            x, y, I0 = self.synthetic_radiograph(size=size, bins=bins, ignore_grid=True)
+    if optical_density:
+        # Generate the null radiograph
+        x, y, I0 = synthetic_radiograph(obj, size=size, bins=bins, ignore_grid=True)
 
-            # Calculate I0 as the mean of the non-zero values in the null
-            # histogram. Zeros are just outside of the illuminate area.
-            I0 = np.mean(I0[I0 != 0])
+        # Calculate I0 as the mean of the non-zero values in the null
+        # histogram. Zeros are just outside of the illuminate area.
+        I0 = np.mean(I0[I0 != 0])
 
-            # Overwrite any zeros in intensity to avoid log10(0)
-            intensity[intensity == 0] = 1
+        # Overwrite any zeros in intensity to avoid log10(0)
+        intensity[intensity == 0] = 1
```

Comment:
Why did you use this approach here?  I feel like this is adding artificial values to the histogram data.  Replacing these points with `np.nan` or just allowing `np.log10` to fill with `-np.inf` seems like the more appropriate route.  To avoid the `RuntimeWarning` in the latter case you can just temporarily activate a warnings filter.

Reply:
Good point: I've taken this out and added a note in the docstring about how to deal with `np.inf` values if necessary. 

This is how I filtered the warnings:

```python
with np.errstate(divide="ignore"):
            intensity = -np.log10(intensity / I0)
```